foundations/src/settings/addr.rs

```diff
@@ -0,0 +1,72 @@
+//! Network address types that support both TCP and Unix domain sockets.
+//!
+//! This module provides the [`ListenAddr`] enum, a flexible address type that can represent
+//! either TCP socket addresses or Unix domain socket paths.
+
+use super::net::SocketAddr;
+use super::Settings;
+use serde::{Deserialize, Serialize};
+use std::fmt;
+
+/// Address that can be either TCP socket or Unix domain socket endpoint
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum ListenAddr {
+    /// TCP network socket address
+    Tcp(SocketAddr),
+    /// Unix domain socket path
+    #[cfg(unix)]
+    Unix(std::path::PathBuf),
+}
+
+impl Default for ListenAddr {
+    fn default() -> Self {
+        let addr: std::net::SocketAddr = (std::net::Ipv4Addr::LOCALHOST, 0).into();
+        ListenAddr::Tcp(addr.into())
+    }
+}
+
+impl From<SocketAddr> for ListenAddr {
+    fn from(addr: SocketAddr) -> Self {
+        ListenAddr::Tcp(addr)
+    }
+}
+
+impl From<std::net::SocketAddr> for ListenAddr {
+    fn from(addr: std::net::SocketAddr) -> Self {
+        ListenAddr::Tcp(addr.into())
+    }
+}
+
+#[cfg(unix)]
+impl From<std::path::PathBuf> for ListenAddr {
+    fn from(path: std::path::PathBuf) -> Self {
+        ListenAddr::Unix(path)
+    }
+}
+
+#[cfg(unix)]
+impl From<&str> for ListenAddr {
```

Comment:
I feel like these impls can be misused and people could end up doing `ListenAddr::from("127.0.0.1:0")`.

Reply:
Fully removed them for now